airflow/jobs.py

```diff
@@ -1089,11 +1090,15 @@ def _process_dags(self, dagbag, dags, tis_out):
             self.logger.info("Processing {}".format(dag.dag_id))
 
             dag_run = self.create_dag_run(dag)
+            models.DagStat.set_dirty(dag.dag_id)
+            dirty_ids.append(dag.dag_id)
```

Comment:
To avoid race conditions, we are not going to implement clean dirty on the BackfillJob.